WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_comments_pagination_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'paginate_comments_links' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+		if ( false !== strpos( $file_content, 'the_comments_navigation' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+		if ( false !== strpos( $file_content, 'the_comments_pagination' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+		if ( false !== strpos( $file_content, 'next_comments_link' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+		if ( false !== strpos( $file_content, 'previous_comments_link' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+	}
+
+	/**
+	 * This functions checks the global content_width is set.
+	 * Checks include enqueue of comment-reply script and useage of the term comment-reply term.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_content_width_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, '$content_width' ) ) {
+			$sniff_helper['content_width'] = true;
+		}
+		if ( false !== strpos( $file_content , '$GLOBALS' . "['content_width']" ) ) {
+			$sniff_helper['content_width'] = true;
+		}
+	}
+
+	/**
+	 * This functions checks if add_editor_style() is called.
+	 * Checks include enqueue of comment-reply script and useage of the term comment-reply term.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_editor_style_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'add_editor_style' ) ) {
+			$sniff_helper['add_editor_style'] = true;
+		}
+	}
+
+	/**
+	 * This functions checks if for get_avatar and wp_list_comments.
+	 * If one is present set flag to true for use later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_avatar_checks( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'get_avatar' ) ) {
+			$sniff_helper['avatar_check'] = true;
+		}
+		if ( false !== strpos( $file_content, 'wp_list_comments' ) ) {
+			$sniff_helper['avatar_check'] = true;
+		}
+	}
+
+	/**
+	 * This functions checks if for wp_nav_menu and register_nav_menu.
+	 * If one is present set flag to true for use later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_custom_menu_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'wp_nav_menu' ) ) {
+			$sniff_helper['custom_menu_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'register_nav_menu' ) ) {
+			$sniff_helper['custom_menu_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'register_nav_menus' ) ) {
```

Comment:
I think there is a way to combine this in the get_theme_supports() function and remove this one. I will do so in the next update.